runtime/tests/org.eclipse.core.tests.harness/src/org/eclipse/core/tests/harness/CoreTest.java

```diff
@@ -55,6 +58,15 @@ public static void debug(String message) {
 		}
 	}
 
+	@Before
+	public void before() throws Exception {
+		setUp();
+	}
+
+	@After
+	public void after() throws Exception {
+		tearDown();
+	}
```

Comment:
I also thought about this and tried to break the "is-a" relation between the modified classes and `TestCase`. 
I had a number of problems with:
* Several usages of methods like `assertXYZ` need to be changed
* Usages of "base" methods (`assertXYZ` / `failXYZ`) that hat been defined in `CoreTest` need to be changed.
* Several test suites are still defined in the old fashion and add tests by calling `junit.framework.TestSuite.addTest(Test)`

The last one is specially tricky because it means the the test suites need to be completely migrated, which is a major undertake.

I saw my approach as a "quick win". I agree with you, it's not pretty, but is a cheap approach and gets the job done.

Can you live with this technical debt? :)

Reply:
Yes, I agree that it will be quite some effort to migrate, in particular because of special test setups like with the session tests.

However, that is independent from where this "cheap approach" is placed. I would still be in favor of having it as local as possible and not placed in the central core test harness plugin.

Reply:
I moved the `before` and `after` methods a bit down in the hierarchy and into `ResourceTest` and `FileSystemTest` (both in the plugin `org.eclipse.core.tests.resources`).